### PR TITLE
Python frontend calls C++ implementation of AC/ Tensorflow

### PR DESCRIPTION
**Pull request by shreyajain17**
To check if this works  -
Also `pip install tensorflow==1.14.0` in the virtual environment 
```model = tc.activity_classifier.create(data, session_id='exp_id', target='activity', _advanced_parameters={'use_tensorflow':True})```

### CODE REVIEW

File: src/python/turicreate/toolkits/activity_classifier/_activity_classifier.py [hunk 1]
```diff
@@ -14,6 +14,7 @@
 import time as _time
 import six as _six
 
+import turicreate as _turicreate
```
Comment (nickjong) [inline src/python/turicreate/toolkits/activity_classifier/_activity_classifier.py]: Why is this needed?

Comment (shreyajain17) [reply src/python/turicreate/toolkits/activity_classifier/_activity_classifier.py]: I am calling _turicreate.extensions in create function.

Comment (nickjong) [reply src/python/turicreate/toolkits/activity_classifier/_activity_classifier.py]: Could we just import it in the `create` function then?

File: src/toolkits/activity_classification/activity_classifier.cpp [hunk 2]
```diff
@@ -637,9 +642,14 @@ std::unique_ptr<data_iterator> activity_classifier::create_iterator(
 }
 
 std::unique_ptr<compute_context>
-activity_classifier::create_compute_context() const
+activity_classifier::create_compute_context(bool use_tensorflow) const
 {
-  return compute_context::create();
+  if (use_tensorflow) {
```
Comment (nickjong) [inline src/toolkits/activity_classification/activity_classifier.cpp]: This could use `read_state("use_tensorflow")` instead of adding an argument to the function, right?

Comment (shreyajain17) [reply src/toolkits/activity_classification/activity_classifier.cpp]: Yes

File: src/python/turicreate/toolkits/activity_classifier/_activity_classifier.py (hunk 1 above)
Comment (shreyajain17) [reply src/python/turicreate/toolkits/activity_classifier/_activity_classifier.py]: I will change it

File: src/ml/neural_net/compute_context.hpp [hunk 3]
```diff
@@ -26,6 +26,10 @@ class compute_context {
   /** Function that yields a compute context. */
   using factory = std::function<std::unique_ptr<compute_context>()>;
 
+  /** Function that yields a compute context (made to return
+   * tf_compute_context). */
+  using tf_factory = std::function<std::unique_ptr<compute_context>()>;
```
Comment (nickjong) [inline src/ml/neural_net/compute_context.hpp]: You don't need a new name here. You can just use `factory` everywhere you were using `tf_factory` since they are both identical aliases for `std::function<std::unique_ptr<compute_context>()>`

File: src/ml/neural_net/compute_context.hpp [hunk 4]
```diff
@@ -34,19 +38,25 @@ class compute_context {
...
     // Registers `factory_fn` at the given priority.
-    registration(int priority, factory factory_fn);
+   registration(int priority, factory factory_fn, tf_factory tf_factory_fn_);
+
+   // Removes the registration. In practice, simplest just not to deallocate...
+   ~registration();
 
-    // Removes the registration. In practice, simplest just not to deallocate...
-    ~registration();
+   int priority() const { return priority_; }
```
Comment (nickjong) [inline src/ml/neural_net/compute_context.hpp]: Did `clang-format` do this, give this an overall indent level of 3 spaces? If so, try giving the `public:` above an indent level of 1 space. I think `clang-format` thinks that stuff should be indented by 1 space relative to `public:`

In short, the number of leading spaces for code should just about always be a multiple of 2, except for `public:`

(See https://google.github.io/styleguide/cppguide.html#Class_Format)

File: src/toolkits/activity_classification/activity_classifier.hpp [hunk 5]
```diff
@@ -14,6 +14,7 @@
 #include <ml/neural_net/compute_context.hpp>
 #include <ml/neural_net/model_backend.hpp>
 #include <ml/neural_net/model_spec.hpp>
+#include <ml/neural_net/tf_compute_context.hpp>
```
Comment (nickjong) [inline src/toolkits/activity_classification/activity_classifier.hpp]: I don't think you need this....
